src/main/java/net/sf/mzmine/modules/peaklistmethods/alignment/join/JoinAlignerTask.java

```diff
@@ -196,16 +223,46 @@ public void run() {
             }
           }
 
+          // compare the similarity of spectra mass lists on MS1 or MS2 level
+          if (compareSpectraSimilarity) {
+            DataPoint[] rowDPs = null;
+            DataPoint[] candidateDPs = null;
+            SpectralSimilarity sim = null;
+
+            // get data points of mass list of the representative scans
+            if (msLevel == 1) {
+              rowDPs =
+                  row.getBestPeak().getRepresentativeScan().getMassList(massList).getDataPoints();
+              candidateDPs = candidate.getBestPeak().getRepresentativeScan().getMassList(massList)
+                  .getDataPoints();
+            }
+
+            // get data points of mass list of the best fragmentation scans
+            if (msLevel == 2) {
+              if (row.getBestFragmentation() != null && candidate.getBestFragmentation() != null) {
+                rowDPs = row.getBestFragmentation().getMassList(massList).getDataPoints();
+                candidateDPs =
+                    candidate.getBestFragmentation().getMassList(massList).getDataPoints();
+              } else
+                continue;
+            }
+
+            // compare mass list data points of selected scans
+            if (rowDPs != null && candidateDPs != null) {
+              sim = createSimilarity(rowDPs, candidateDPs);

```

Comment:
So, I see that _sim_ is assigned a value here, but I don't see where this value is used? In other words, where in the code do you actually take the calculated spectral similarity into account?